packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts

```diff
@@ -0,0 +1,106 @@
+import Vuex, { Store } from 'vuex';
+import { createLocalVue } from '@vue/test-utils';
+import { SearchRequest } from '@empathyco/x-types';
+import { getSearchResponseStub } from '../../../../__stubs__/search-response-stubs.factory';
+import { getMockedAdapter, installNewXPlugin } from '../../../../__tests__/utils';
+import { SafeStore } from '../../../../store/__tests__/utils';
+import { queriesPreviewXStoreModule } from '../module';
+import {
+  QueriesPreviewState,
+  QueriesPreviewGetters,
+  QueriesPreviewMutations,
+  QueriesPreviewActions
+} from '../types';
+
+describe('testing queries preview module actions', () => {
+  const mockedSearchResponse = getSearchResponseStub();
+  const getQueryPreviewRequest = (query: string): SearchRequest => ({
+    query,
+    rows: 3,
+    extraParams: {
+      extraParam: 'extra param'
+    }
+  });
+
+  const adapter = getMockedAdapter({
+    search: mockedSearchResponse
+  });
+
+  const localVue = createLocalVue();
+  localVue.config.productionTip = false; // Silent production console messages.
+  localVue.use(Vuex);
+
+  const store: SafeStore<
+    QueriesPreviewState,
+    QueriesPreviewGetters,
+    QueriesPreviewMutations,
+    QueriesPreviewActions
+  > = new Store(queriesPreviewXStoreModule as any);
+  installNewXPlugin({ adapter, store }, localVue);
```

Comment:
You are creating a global store, and you never reset its state. Therefore each test case result is dependant from the previous test. You can either do something similar like we do with component render factories, or reset the store state before every test.

Reply:
You're right! I actually added the util to reset it but forgot to use it in the tests 😅. I've added it!